src/BuiltInTools/HotReloadClient/Utilities/ArrayBufferWriter.cs

```diff
@@ -0,0 +1,269 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#if !NET
+
+#nullable enable
+
+using System;
+using System.Buffers;
+using System.Diagnostics;
+using System.Runtime.CompilerServices;
+
+namespace System.Buffers;
+
+/// <summary>
+/// Represents a heap-based, array-backed output sink into which <typeparam name="T"/> data can be written.
+/// </summary>
+internal sealed class ArrayBufferWriter<T> : IBufferWriter<T>
```

Comment:
This is copied from the runtime, right? If so, consider adding a comment with a permalink to the original file to indicate where it came from.